<h1 style="text-align:center; color:whitee;"> Les Fondamentaux du language python </h1>

| Concept fondamental      | Exemple en Python                  | Description                                      |
|-------------------------|------------------------------------|--------------------------------------------------|
| Variable                | `x = 5`                            | Stocke une valeur                                |
| Type de données         | `int`, `float`, `str`, `bool`      | Nombres, texte, booléens                         |
| Opérations              | `+`, `-`, `*`, `/`, `%`            | Calculs mathématiques et opérations sur les données|
| Conditionnelles         | `if`, `elif`, `else`               | Exécute du code selon une condition              |
| Boucles                 | `for`, `while`                     | Répète des instructions                          |
| Fonctions               | `def ma_fonction():`               | Regroupe du code réutilisable                    |
| Listes                  | `ma_liste = [1, 2, 3]`             | Tableau de valeurs modifiable                    |
| Dictionnaires           | `d = {"nom": "Alice", "age": 20}`  | Stocke des paires clé/valeur                     |
| Importation de modules  | `import math`                      | Utilise des bibliothèques

## Les Variables 

In [1]:
sexe = "F"
print(sexe)

F


In [7]:
sexe = "H"
age = 25
nom = "John"
print(f"je m'appelle {nom}. Je suis de sexe {sexe} et j'ai {age} ans.")

je m'appelle John. Je suis de sexe H et j'ai 25 ans.


### L’interpolation de chaînes en Python

L’interpolation permet d’insérer des valeurs de variables directement dans une chaîne de caractères.  
Voici les méthodes principales :

| Méthode                | Exemple                                      | Résultat                |
|------------------------|----------------------------------------------|-------------------------|
| f-string (recommandé)  | `name = "Alice"`<br>`f"Bonjour {name}!"`     | Bonjour Alice!          |
| format()               | `"Bonjour {}!".format(name)`                 | Bonjour Alice!          |
| % (ancienne méthode)   | `"Bonjour %s!" % name`                       | Bonjour Alice!          |

**Exemple avec f-string :**
```python
age = 20
print(f"J'ai {age} ans")
# Affiche : J'ai 20 ans

In [2]:
age  = 40
print(f"j'ai {age} ans")

j'ai 40 ans


### Les types de données en Python

- **int** : nombres entiers  
  Exemple : `x = 5`

- **float** : nombres décimaux  
  Exemple : `y = 3.14`

- **str** : chaînes de caractères (texte)  
  Exemple : `nom = "Alice"`

- **bool** : booléens (vrai ou faux)  
  Exemple : `actif = True`

- **list** : liste d’éléments  
  Exemple : `notes = [12, 15, 18]`

- **tuple** : liste non modifiable  
  Exemple : `coord = (10, 20)`

- **dict** : dictionnaire (paires clé/valeur)  
  Exemple : `personne = {"nom": "Bob", "age": 25}`

- **set** : ensemble d’éléments uniques  
  Exemple : `couleurs = {"rouge", "vert", "bleu"}`

In [3]:
coord = (10, 20)
print(f"coordonnées : {coord[0]} {coord[1]}")

coordonnées : 10 20


### Détermination du type de données en Python

Pour connaître le type d’une variable en Python, on utilise la fonction `type()` :

```python
x = 5
print(type(x))  # <class 'int'>

y = 3.14
print(type(y))  # <class 'float'>

nom = "Alice"
print(type(nom))  # <class 'str'>

notes = [12, 15, 18]
print(type(notes))  # <class 'list'>
```

In [8]:
collection = [1, 2, 3, 4, "nom", "prénom", 3.14, True]
print(len(collection))

8


### Opérations dans Python

- **Addition** : `a + b`  
  Exemple : `3 + 2` donne `5`

- **Soustraction** : `a - b`  
  Exemple : `5 - 1` donne `4`

- **Multiplication** : `a * b`  
  Exemple : `4 * 2` donne `8`

- **Division** : `a / b`  
  Exemple : `10 / 2` donne `5.0`

- **Division entière** : `a // b`  
  Exemple : `7 // 2` donne `3`

- **Modulo (reste de division)** : `a % b`  
  Exemple : `7 % 2` donne `1`

- **Puissance** : `a ** b`  
  Exemple : `2 ** 3` donne

In [9]:
variable = 0.1 + 0.2
print(variable)

0.30000000000000004


L’addition `0.1 + 0.2` en Python donne `0.30000000000000004` à cause des limites de la représentation des nombres à virgule flottante.

Pour afficher un résultat arrondi, utilise la fonction `round()` :

```python
variable = 0.1 + 0.2
print(round(variable, 2))  # Affiche 0.3
```

In [13]:
valeur = 0.1 + 0.2
print(round(valeur,2)
      )


0.3


In [16]:
print(3//2)

1


### Comparaison en Python

Les opérateurs de comparaison permettent de comparer des valeurs. Ils renvoient un booléen (`True` ou `False`).

| Opérateur | Signification        | Exemple         | Résultat      |
|-----------|---------------------|-----------------|--------------|
| `==`      | Égal à              | `3 == 3`        | `True`       |
| `!=`      | Différent de        | `3 != 4`        | `True`       |
| `>`       | Supérieur à         | `5 > 2`         | `True`       |
| `<`       | Inférieur à         | `2 < 5`         | `True`       |
| `>=`      | Supérieur ou égal   | `5 >= 5`        | `True`       |
| `<=`      | Inférieur ou égal   | `3 <= 4`        | `True`       |

**Exemple :**
```python
a = 10
b = 5
print(a > b)      # True
print(a == b)     # False
print(a != b)     #

In [18]:
print(True == True)

True


### Conditionnelles en Python

Les instructions conditionnelles permettent d’exécuter du code seulement si une condition est vraie.

**Syntaxe de base :**
```python
if condition:
    # code exécuté si la condition est vraie
elif autre_condition:
    # code exécuté si la première condition est fausse mais celle-ci vraie
else:
    # code exécuté si aucune condition n’est vraie
```

**Exemple :**
```python
age = 18
if age >= 18:
    print("Majeur")
else:
    print("Mineur")
```